enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion/context.tsx

```diff
@@ -119,40 +125,58 @@ export const InteractiveQuestionProvider = ({
     loadQuestion();
   };
 
-  const { isRunning: isQueryRunning } = uiControls;
-
   useUnmount(() => {
     dispatch(resetQB());
   });
+  return { resetQuestion, isQuestionLoading };
+};
 
-  const defaultHeight = card ? getDefaultVizHeight(card.display) : undefined;
+export const InteractiveQuestionProvider = ({
+  children,
+  location,
+  params,
+  componentPlugins,
+  onReset,
+  onNavigateBack,
+  withTitle = false,
+  customTitle,
+  withResetButton,
+}: PropsWithChildren<
+  UseLoadQuestionParams & {
+    componentPlugins?: SdkPluginsConfig;
+    withResetButton?: boolean;
+    onReset?: () => void;
+    onNavigateBack?: () => void;
+    withTitle?: boolean;
+    customTitle?: ReactNode;
+  }
+>) => {
+  const { isQuestionLoading, resetQuestion } = useLoadQuestion({
+    location: location,
+    params: params,
+  });
 
-  const plugins = componentPlugins || globalPlugins;
-  const mode = question && getEmbeddingMode(question, plugins || undefined);
+  const { question } = useInteractiveQuestionData();
 
-  if (question) {
-    question.alertType = returnNull; // FIXME: this removes "You can also get an alert when there are some results." feature for question
-  }
+  const globalPlugins = useSdkSelector(getPlugins);
+  const plugins = componentPlugins || globalPlugins;
+  const mode = useMemo(
+    () => question && getEmbeddingMode(question, plugins || undefined),
+    [plugins, question],
+  );
 
   return (
     <InteractiveQuestionContext.Provider
       value={{
-        question,
-        card,
-        result,
-        uiControls,
-        queryResults,
-        plugins,
-        mode,
-        defaultHeight,
         isQuestionLoading,
-        isQueryRunning,
         resetQuestion,
         onReset: onReset || resetQuestion,
         onNavigateBack,
         withTitle,
         customTitle,
-        withResetButton: hasQuestionChanges && withResetButton,
+        withResetButton,
```

Comment:
I see `hasQuestionChanges` is now removed. Was that a mistake?

Reply:
I've moved that to the `useInteractiveQuestionData` hook - I figured that we should use that within the `ResetButton` component and so that any changes to `hasQuestionChanges` (which depends on the redux `card`) doesn't cause the whole context to reload